dateparser/date_parser.py

```diff
@@ -72,7 +52,7 @@ def get_period(res):
             ('month', ['month']),
             ('year', ['year']),
         ])
-        for period, markers in periods.iteritems():
+        for period, markers in six.iteritems(periods):
             for marker in markers:
                 if getattr(res, marker) is not None:
```

Comment:
result object has attributes with values set to None, hence, using getattr and check for value is needed here. Since, `weekday` can be `0`, `is not None` predicate is used.